packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js

```diff
@@ -235,12 +265,34 @@ SelfUtils.isAdmittedGuest = (oldSelf, changedSelf) => {
   return SelfUtils.isLocusGuestUnadmitted(oldSelf) && SelfUtils.isLocusGuestAdmitted(changedSelf);
 };
 
-SelfUtils.mutedByOthers = (oldSelf = {}, changedSelf) => {
+SelfUtils.mutedByOthersChanged = (oldSelf = {}, changedSelf) => {
   if (!changedSelf) {
     throw new ParameterError('New self must be defined to determine if self was muted by others.');
   }
 
-  return changedSelf.muted && (oldSelf.lastModified !== changedSelf.lastModified || oldSelf.modifiedBy !== changedSelf.modifiedBy);
+  if (!oldSelf && !changedSelf.remoteMuted) {
+    // we don't want to be sending the 'meeting:self:unmutedByOthers' notification on meeting join
+    return false;
+  }
+
+  return (oldSelf.remoteMuted !== changedSelf.remoteMuted) || (changedSelf.remoteMuted && (oldSelf.unmuteAllowed !== changedSelf.unmuteAllowed));
+};
+
+SelfUtils.requestedToUnmuteReceived = (oldSelf = {}, changedSelf) => {
+  if (!changedSelf) {
+    throw new ParameterError('New self must be defined to determine if self received request to unmute.');
+  }
+
+  return changedSelf.requestedToUnmute && !oldSelf.requestedToUnmute;
+};
+
+
+SelfUtils.localAudioUnmuteRequiredChanged = (oldSelf = {}, changedSelf) => {
+  if (!changedSelf) {
+    throw new ParameterError('New self must be defined to determine if localAudioUnmuteRequired changed.');
+  }
+
+  return !changedSelf.remoteMuted && changedSelf.localAudioUnmuteRequired && (oldSelf.localAudioUnmuteRequired !== changedSelf.localAudioUnmuteRequired);
```

Comment:
so, when both "hard mute" and "moderated unmute mode" features are used and host mutes you, Locus sends us muted=true together with disallowUnmute=true. Then later if the host unmutes you, we receive localAudioUnmuteRequired=true and muted=false, but disallowUnmute stays still as true.
I've just realised that as a result, in this scenario we will send the event 'meeting:self:unmutedByOthers' twice:
one because of audio.muted changing from true to false and the other because of us doing the local unmute. So I'm thinking now that maybe we should have a separate event for the localAudioUnmuteRequired - something like 'meeting:self:locallyUnmutedByOthers'

Reply:
I've decided that 'meeting:self:unmutedByOthers' should be sent only when we get remotely unmuted and we are also locally unmuted. This will avoid having duplicate notification I mention in previous comment and also makes sense as you are not really unmuted if you have locally muted yourself and server toggles "remote mute".